src/python/pants/base/specs.py

```diff
@@ -388,6 +388,35 @@ def resolved_files(self) -> Tuple[str, ...]:
         return self._snapshot.files
 
 
+@dataclass(frozen=True)
+class FilesystemMergedSpec(FilesystemResolvedSpec):
+    """When multiple file arguments are given that resolve to the same owning target, this type
```

Comment:
We generally like the first sentence of a docstring to fit on a single line, per PEP 257. 

This is also pretty circuitous text. Something like this is sufficient as a description of the type itself:

```
A newtype representing multiple filesystem specs belonging to the same target.
```